docs/design/2019-11-14-tidb-builtin-diagnostics-zh_CN.md

```diff
@@ -0,0 +1,680 @@
+| pd   | 10.0.1.23:2379  | 2019/11/01 15:04:00.033 | WARN  | ...                                |
+| tikv | 10.0.1.24:20160 | 2019/11/01 15:05:00.033 | ERROR | ...                                |
+| tikv | 10.0.1.24:20160 | 2019/11/01 15:05:00.033 | ERROR | ...                                |
+| tikv | 10.0.1.24:20160 | 2019/11/01 15:05:00.033 | ERROR | ...                                |
++------+-----------------+-------------------------+-------+------------------------------------+
+9 rows in set (0.00 sec)
+
+mysql> select * from tidb_cluster_log where type = 'pd';
++------+----------------+-------------------------+-------+---------+
+| type | address        | time                    | level | message |
++------+----------------+-------------------------+-------+---------+
+| pd   | 10.0.1.23:2379 | 2019/11/01 15:04:00.033 | WARN  | ...     |
+| pd   | 10.0.1.23:2379 | 2019/11/01 15:04:00.033 | WARN  | ...     |
+| pd   | 10.0.1.23:2379 | 2019/11/01 15:04:00.033 | WARN  | ...     |
++------+----------------+-------------------------+-------+---------+
+3 rows in set (0.00 sec)
+
+mysql> select * from tidb_cluster_log where level = 'ERROR';
++------+-----------------+-------------------------+-------+---------+
+| type | address         | time                    | level | message |
++------+-----------------+-------------------------+-------+---------+
+| tikv | 10.0.1.24:20160 | 2019/11/01 15:05:00.033 | ERROR | ...     |
+| tikv | 10.0.1.24:20160 | 2019/11/01 15:05:00.033 | ERROR | ...     |
+| tikv | 10.0.1.24:20160 | 2019/11/01 15:05:00.033 | ERROR | ...     |
++------+-----------------+-------------------------+-------+---------+
+3 rows in set (0.00 sec)
+
+mysql> select * from tidb_cluster_log where message like '%table%';
++------+----------------+-------------------------+-------+------------------------------------+
+| type | address        | time                    | level | message                            |
++------+----------------+-------------------------+-------+------------------------------------+
+| tidb | 127.0.0.1:4000 | 2019/11/01 15:03:00.033 | INFO  | [BIG_TXN] [conn=10] [table id] ... |
+| tidb | 127.0.0.1:4000 | 2019/11/01 15:03:00.033 | INFO  | [BIG_TXN] [conn=10] [table id] ... |
+| tidb | 127.0.0.1:4000 | 2019/11/01 15:03:00.033 | INFO  | [BIG_TXN] [conn=10] [table id] ... |
++------+----------------+-------------------------+-------+------------------------------------+
+3 rows in set (0.00 sec)
+```
+
+### 集群诊断
+
+在当前的集群拓扑下，各个组件分散，数据源和数据格式异构，不便于通过程序化的手段进行集群诊断，所以需要人工进行问题诊断。通过前面几层提供的数据系统表，每一个 TiDB 节点都有了一个稳定的全集群 Global View，所以可以在这个基础上实现一个问题诊断框架。通过定义诊断规则能够快速发现集群的已有问题和潜在问题。
+
+**诊断规则定义**：诊断规则是通过读入各个系统表的数据，并通过检测异常数据发现问题的逻辑。
+
+诊断规则可以分为三个层次：
+
+- 发现潜在问题：比如通过判断磁盘容量和磁盘使用量的比例发现磁盘容量不足
+- 发现一有问题：比如通过查看负载情况，发现 Coprocessor 的线程池已经跑满
+- 给出修复建议：比如通过分析磁盘 IO 发现延迟过高，可以给出更换磁盘的建议
+
+本提案主要负责实现诊断框架和部分诊断规则，更多的诊断规则需要根据使用经验逐步沉淀，最终形成一个专家系统，降低使用门槛和运维难度。后续内容不详细探讨具体某条的诊断规则，主要聚焦诊断框架的实现。
+
+#### 诊断框架设计
+
+诊断框架的设计需要考虑多种用户使用场景，包括不限于：
```

Comment:
Could we make the diagnostics framework to be standalone script?

Reply:
Yes, we can make it as a shell script and execute it in external.

Reply:
Or we can remote this session in this RFC. It can be defined in the next stage.

Reply:
I think use shell as the diagnostics rule is a good idea, I will add to this proposal and keep the others options.